src/deluge/gui/ui/keyboard/chords.h

```diff
@@ -0,0 +1,170 @@
+const Chord k9 = {"9",
+                  {{ROOT, MAJ3, P5, MIN7, MAJ9, NONE, NONE},
+                   {ROOT, MAJ3 + OCT, P5, MIN7, MAJ9, NONE, NONE},
+                   {ROOT, MAJ3 + OCT, P5, MIN7 + OCT, MAJ9, NONE, NONE}}};
+const Chord kM9 = {"M9",
+                   {{ROOT, MAJ3, P5, MAJ7, MAJ9, NONE, NONE},
+                    {ROOT, MAJ3 + OCT, P5, MAJ7, MAJ9, NONE, NONE},
+                    {ROOT, MAJ3 + OCT, P5, MAJ7 + OCT, MAJ9, NONE, NONE}}};
+const Chord kMinor9 = {"-9",
+                       {{ROOT, MIN3, P5, MIN7, MAJ9, NONE, NONE},
+                        {ROOT, MIN3 + OCT, P5, MIN7, MAJ9, NONE, NONE},
+                        {ROOT, MIN3 + OCT, P5, MIN7 + OCT, MAJ9, NONE, NONE}}};
+const Chord k11 = {"11",
+                   {{ROOT, MAJ3, P5, MIN7, MAJ9, P11, NONE},
+                    {ROOT, MAJ3 + OCT, P5, MIN7, MAJ9, P11, NONE},
+                    {ROOT, MAJ3 + OCT, P5, MIN7 + OCT, MAJ9, P11, NONE}}};
+const Chord kM11 = {"M11",
+                    {{ROOT, MAJ3, P5, MAJ7, MAJ9, P11, NONE},
+                     {ROOT, MAJ3 + OCT, P5, MAJ7, MAJ9, P11, NONE},
+                     {ROOT, MAJ3 + OCT, P5, MAJ7 + OCT, MAJ9, P11, NONE}}};
+const Chord kMinor11 = {"-11",
+                        {{ROOT, MIN3, P5, MIN7, MAJ9, P11, NONE},
+                         {{ROOT, P4, MIN7, MIN3 + OCT, P5 + OCT, NONE, NONE}, "SO WHAT"},
+                         {ROOT, MIN3 + OCT, P5, MIN7, MAJ9, P11, NONE},
+                         {ROOT, MIN3 + OCT, P5, MIN7 + OCT, MAJ9, P11, NONE}}};
+// 11th are often omitted in 13th chords because they clash with the major 3rd
+// if anything, the 11th is often played as a #11
+const Chord k13 = {"13",
+                   {{ROOT, MAJ3, P5, MIN7, MAJ9, MAJ13, NONE},
+                    {ROOT, MAJ3 + OCT, P5, MIN7, MAJ9, MAJ13, NONE},
+                    {ROOT, MAJ3 + OCT, P5, MIN7 + OCT, MAJ9, MAJ13, NONE}}};
+const Chord kM13 = {"M13",
+                    {{ROOT, MAJ3, P5, MAJ7, MAJ9, MAJ13, NONE},
+                     {ROOT, MAJ3 + OCT, P5, MAJ7, MAJ9, MAJ13, NONE},
+                     {ROOT, MAJ3 + OCT, P5, MAJ7 + OCT, MAJ9, MAJ13, NONE}}};
+const Chord kMinor13 = {"-13",
+                        {{ROOT, MIN3, P5, MIN7, MAJ9, P11, MAJ13},
+                         {ROOT, MIN3 + OCT, P5, MIN7, MAJ9, P11, MAJ13},
+                         {ROOT, MIN3 + OCT, P5, MIN7 + OCT, MAJ9, P11, MAJ13}}};
+
+/// @brief A collection of chords
+class ChordList {
+public:
+	ChordList();
+
+	/**
+	 * @brief Get a voicing for a chord with a given index. If the voicingOffset for that Chord is out of bounds,
+	 * return the max or min voicing depending on the direction.
+	 *
+	 * @param chordNo The index of the chord
+	 * @return The voicing
+	 */
+	Voicing getChordVoicing(int32_t chordNo);
+
+	void adjustChordRowOffset(int32_t offset);
+	void adjustVoicingOffset(int32_t chordNo, int32_t offset);
+
+	Chord chords[kUniqueChords];
+	int32_t voicingOffset[kUniqueChords] = {0};
+	uint32_t chordRowOffset = 0;
```

Comment:
Would it be better put these two state trackers in the `KeyboardStateChord` struct or here? If they are kept there, then we could more easily reuse the `ChordList` class for the chord columns if we want to add more functionality to those. If that isn't important, then just unifying those chord columns with these chord definitions, without the `ChordList` class could be enough. Thoughts?

Reply:
I'd go with your gut feel. Either way it doesn't look super difficult to change in the future if needed, and I think you might be the most likely person to extend it into chord columns anyway!